### PR TITLE
Utopia 210: Init NestJS API Project

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- Update node 16 to 18.9 for consistency
- Added Swagger
- Initialize NestJS to Express base
- Added .env inside `/src/backend`
```
ENV = development
PORT = 3000

API_NAME = Digital Privacy Impact Assessment API
API_VERSION = 0.0.1
```
- Added dependencies: 
```
"@nestjs/swagger": "6.1.2",
"dotenv": "16.0.2",
"express": "4.18.1",
"morgan": "1.10.0",
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [x] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [x] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [x] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [x] My changes includes Swagger documentation updates that reflect the changes I am introducing

image

### CODE REVIEW

**Comment by kushal-arora-fw** [comment]
@akroon3r - are we allowing only 25lines per file? 🤔 🤔 Wondering if just those would be enough. Thoughts anyone else?

image

File: src/backend/src/app.config.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            : ['log', 'warn', 'error'],
+      },
+    );
+
+  // append request logs
+  app.use(morgan('tiny'));
+
+  app.setGlobalPrefix('/api');
+
+  // Enable Swagger for non prod environment
```
Comment (akroon3r) [inline src/backend/src/app.config.ts:37]: If we're developing a public API, I'd love for our swagger to be readily available in PROD.

What do you think?

Comment (kushal-arora-fw) [reply src/backend/src/app.config.ts:37]: I mean, sure if we need it to be public. Are you talking from a sense that this repo is public and so people should have access to the API documentation too? If so, sure - I will go ahead and make that change 👍 You have better context than me.

However, in that case if we are having our APIs and documentation public - we should have API throttling set up - where someone maliciously do not try to bring our system down by continuously hitting the endpoints / bots. Thoughts?

**Comment by kushal-arora-fw** [comment]
> 



> @akroon3r - are we allowing only 25lines per file? 🤔 🤔 Wondering if just those would be enough. Thoughts anyone else?
> 
> image
> 
> Good catch, and thank you for checking out CC. I didn't really get a chance to actually configure code climate. We can go through this as a team if you want?
> 
> Sidenote: Failing actions are not blockers YET. We can work on establishing this as a group after we've got outside of the init work we're doing right now.

Sure, we can chat over it 👍

**Comment by akroon3r** [comment]
@kushal-arora-fw just need to point out that I really love what you did in the PR outlining what dependencies you added up front before I read the diff.

I think this is an awesome precedence to set for the group. @binggaofw @Bottle7 @antsand What do you guys think? Is this something we can agree to uphold for the development lifecycle going forwards?

**Review by binggaofw — APPROVED** [review]

**Review by akroon3r — APPROVED** [review]

**Pull request merged**